pkg/apis/aws/types_worker.go

```diff
@@ -117,3 +119,24 @@ const (
 	// VolumeTypeGP3 is a constant for the gp3 volume type.
 	VolumeTypeGP3 VolumeType = "gp3"
 )
+
+// HTTPTokensValue is a constant for HTTPTokens values.
+type HTTPTokensValue string
+
+const (
+	// HTTPTokensRequired is a constant for requiring the use of tokens to access IMDS. Effectively disables access via
+	// the IMDSv1 endpoints.
+	HTTPTokensRequired HTTPTokensValue = "required"
+	// HTTPTokensOptional that makes the use of tokens for IMDS optional. Effectively allows access via both IMDSv1 and
+	// IMDSv2 endpoints.
+	HTTPTokensOptional HTTPTokensValue = "optional"
+)
+
+// InstanceMetadataOptions contains configuration for controlling access to the metadata API.
+type InstanceMetadataOptions struct {
```

Comment:
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-launchtemplate-launchtemplatedata-metadataoptions.html also provides another option to enable/disable access to instance tags via `InstanceMetadataTags`. By default it is disabled. Should this also be configurable?

Reply:
This wouldn't work at the moment because the MCM side would need to be adapted as well.

Reply:
But as with many of these requests, we add features as requested. The main focus has been controlling the access to IMDS for this one. 

We can go back and add the flags to MCM and then the provider extension in a separate PR when needed (e.g. like the ipv6 flag), unless you think that they have to be included in this PR.

Reply:
We can always enhance the struct with additional fields at a later point in time.